private/bufpkg/bufimage/bufimageutil/bufimageutil_test.go

```diff
@@ -237,3 +256,75 @@ func imageIsDependencyOrdered(image bufimage.Image) bool {
 	}
 	return true
 }
+
+var benchmarkCases = []*struct {
+	folder string
+	image  bufimage.Image
+	types  []string
+}{
+	{
+		folder: "testdata/extensions",
+		types:  []string{"pkg.Foo"},
+	},
+	{
+		folder: "testdata/importmods",
+		types:  []string{"ImportRegular", "ImportWeak", "ImportPublic", "NoImports"},
+	},
+	{
+		folder: "testdata/nesting",
+		types:  []string{"pkg.Foo", "pkg.Foo.NestedFoo.NestedNestedFoo", "pkg.Baz", "pkg.FooEnum"},
+	},
+	{
+		folder: "testdata/options",
+		types:  []string{"pkg.Foo", "pkg.FooEnum", "pkg.FooService", "pkg.FooService.Do"},
+	},
+}
+
+func TestMain(m *testing.M) {
+	logger, err := zap.NewDevelopment()
+	if err != nil {
+		_, _ = fmt.Fprintf(os.Stderr, "failed to create logger: %v", err)
+		os.Exit(1)
+	}
+	ctx := context.Background()
+	for _, benchmarkCase := range benchmarkCases {
+		_, image, err := getImage(ctx, logger, benchmarkCase.folder)
+		if err != nil {
+			_, _ = fmt.Fprintf(os.Stderr, "failed to create image for %q: %v", benchmarkCase.folder, err)
+			os.Exit(1)
+		}
+		benchmarkCase.image = image
+	}
```

Comment:
Yep, I'll fix it. I originally did it this way, to avoid any other stuff in the benchmark function. And then I realized that filtering an image was destructive, and that we _have_ to make copies of the image, so I added the `StopTimer`/`StartTimer` stuff but never moved the rest of this back in.

Reply:
Fixed